fud/fud/stages/interpreter.py

```diff
@@ -44,60 +42,67 @@ def __init__(
             target_state=output_name,
             input_type=SourceType.Stream,
             output_type=output_type,
-            config=config,
             description=desc,
         )
 
         self.flags = flags
         self.debugger_flags = debugger_flags
-        self.data_path = self.config["stages", "verilog", "data"]
-
-        self.setup()
-
-    def _define_steps(self, input_data):
-
-        cmd = " ".join(
-            [
-                self.cmd,
-                self.flags,
-                unwrap_or(self.config["stages", self.name, "flags"], ""),
-                "-l",
-                self.config["global", "futil_directory"],
-                "--data" if self.data_path else "",
-                "{data_file}" if self.data_path else "",
-                "{target}",
-                "debug" if self.target_stage == _DEBUGGER_TARGET else "",
-                self.debugger_flags if self.target_stage == _DEBUGGER_TARGET else "",
-                unwrap_or(self.config["stages", self.name, "debugger", "flags"], "")
-                if self.target_stage == _DEBUGGER_TARGET
+
+    def _is_debugger(self):
+        """
+        Am I a debugger?
+        """
+        return self.target_state == _DEBUGGER_TARGET
+
+    def _define_steps(self, input_data, builder, config):
+
+        script = config["stages", self.name, "exec"]
+        data_path = config["stages", "verilog", "data"]
+
+        cmd = [
+            script,
+            self.flags,
+            unwrap_or(config["stages", self.name, "flags"], ""),
+            "-l",
+            config["global", "futil_directory"],
+            "--data" if data_path else "",
+            "{data_file}" if data_path else "",
+            "{target}",
+        ]
+
+        if self._is_debugger():
+            cmd += [
+                "debug" if self._is_debugger() else "",
+                self.debugger_flags if self._is_debugger() else "",
+                unwrap_or(config["stages", self.name, "debugger", "flags"], "")
+                if self._is_debugger()
```

Comment:
Shouldn't be necessary to check `self.is_debugger()` again below, since its already wrapped in the conditional.